java-checks/src/main/java/org/sonar/java/checks/tests/MockitoAnnotatedObjectsShouldBeInitializedCheck.java

```diff
@@ -154,16 +155,33 @@ private static boolean isMockitoJUnitRuleInvoked(ClassTree clazz) {
     for (VariableTree field : collected) {
       if (field.type().symbolType().is("org.mockito.junit.MockitoRule")) {
         ExpressionTree initializer = field.initializer();
+        Symbol symbol = field.symbol();
         if (initializer != null && initializer.is(Tree.Kind.METHOD_INVOCATION) &&
-          MOCKITO_JUNIT_RULE.matches((MethodInvocationTree) initializer) &&
-          field.symbol().metadata().isAnnotatedWith(RULE_ANNOTATION)) {
+          symbol.metadata().isAnnotatedWith(RULE_ANNOTATION) &&
+          symbol.isPublic() &&
+          isInitializedWithRule((MethodInvocationTree) initializer)) {
           return true;
         }
       }
     }
     return false;
   }
 
+  private static boolean isInitializedWithRule(MethodInvocationTree mit) {
+    MethodInvocationTree current = mit;
+    while (true) {
+      if (MOCKITO_JUNIT_RULE.matches(current)) {
+        return true;
+      }
+      MemberSelectExpressionTree memberSelect = (MemberSelectExpressionTree) current.methodSelect();
```

Comment:
How do you know it's not simply a method invocation without a member select!?

The following code makes the rule crash:
```java
  public class FooTest3 {
    @Rule
    public MockitoRule rule = foo();

    @Mock // Compliant
    private Bar bar;

    private MockitoRule foo() { return null; }
  }
```

Reply:
I will integrate that